salt/metalk8s/kubeadm/init/certs/init.sls

```diff
@@ -0,0 +1,14 @@
+#
+# State for kubeadm init certs phase.
+#
+# The goal is to have all the CA and certs for all services
+#
+# Available states
+# ================
+#
+# * ca        -> create the root CA server and advertise it
+# * apiserver -> create the kube API server certificates using root CA
+
+include:
```

Comment:
Now this is a bit of an odd top-level state. I don't think we really want this one. The CA at least should be set up on one specific machine only. This is not in line with the `kubeadm` flow, though that's OK: `kubeadm` isn't multi-node aware, and would require the CAs to be copied to all nodes, or certs to be signed out-of-band, so in the end there's one CA only.

Reply:
Yes agree, but for me (like I already said several time in slack) kubeadm approach is totaly NOT idempotent and it's totaly different from salt approach and keep the repo look like kubeadm is maybe not a good idea but it's only my opinion.

Reply:
I agree with you :D In general, we take (for now) the kubeadm flow as a guideline. However, when there's parts of said flow that really don't match our needs, we can deviate, no problem whatsoever!

Reply:
Let's even remove this include since we won't call directly the `init` state.

Reply:
For me I would add only CAs in the init.sls as CAs it's really the "initialization of the certs" make sense ?

Reply:
Yeah but running this on another node you would think "this is initializing the cert on this server based on the CA already set up". So it's kind of risky. But since all this will be automated, it's fine by me.
Do as you wish :)

Reply:
I will keep it like that for the moment to not have a init who include nothing and we will update it later on if needed